cmd/html/script.js

```diff
@@ -0,0 +1,200 @@
+        .paddingRight(2)
+        .paddingBottom(2)
+        .paddingLeft(2)
+        .round(2)
+        // .padding(0)
+        .tile(d3.treemapSquarify.ratio(1))(root)
+
+        console.log("Treemap",treemap)
+        // console.log(treemap.leaves())    
+        
+    const g = svg.append("g")
+    .attr("transform", "translate(" + treemap.leaves()[0].x0 + "," + treemap.leaves()[0].y0 + ")");
+    
+
+    
+
+    // add nodes
+    g.selectAll("a")
+    .data(treemap.descendants().slice(0))
+    .enter()
+    .append("a")
+    .attr("href", (d) => {
+      if (d.depth === 0) {
+        return "#"
+      }
+      return d.height === 1 ? "node/"+d.data.name : "pod/"+d.data.name
+      return d.data.name+"/"
+    
+    })
+
+    // add rectangles for each pod
+    g.selectAll("a")
+    .data(treemap.descendants().slice(0))
+    .append("rect")
+    .attr("x", (d) => {
+        // console.log(d.x0, d.x1);
+        // console.log(d)
+        return d.x0-0;
+    })
+    .attr("y", (d) => {
+        // console.log(d.y0, d.y1);
+        return d.y0-0;
+    })
+    .attr("width", (d) => (d.x1 - d.x0)+0)
+    .attr("height", (d) => (d.y1 - d.y0)+0)
+    .style("fill", (d) => {
+        // console.log("DATA type:", d);
+        if (d.depth === 0) {
+            return "lightblue"
+        }
+        return d.height === 1 ? "#03385c" : "#3568a3"
+    })
+    .attr("margin", (d) => {
+        if (d.depth === 0) {
+            return "0px"
+        }
+        return d.height === 1 ? "5px" : "10px"
+    });
+    
+    // add text for each pod
+    // g.selectAll("a")
+    // .data(treemap.descendants().slice(0))
+    // // .enter()
+    // .append("text")
+    // .attr("x", (d) => d.x0 + 10)
+    // .attr("y", (d) => {
+    //     if (d.depth === 0) {
+    //         return "0"
+    //     }
+    //     return d.height === 1 ? d.y0 + 20 : d.y0 + 40
+    // })
+    // .attr("font-size", "14px")
+    // .attr("fill", "white")
+    // .text((d) => d.data.name)
+    
+    // print final treemap
+    console.log("Final treemap",treemap)
+
+    // add a title for each pod
+    g.selectAll("a")
+    .data(treemap.descendants().slice(0))
+    // .enter()
+    .append("title")
+    .text((d) => d.data.name);
+
+    g.selectAll("a")
+    .data(treemap.descendants().slice(0))
+    // .enter()
+    .append("label")
+    .text((d) => d.data.name);
+
+
+    // g.append("text")
+    //     .attr("clip-path", (d, i) => `url(${new URL(`#${uid}-clip-${i}`, location)})`)
+    //   .selectAll("tspan")
+    //   .data(treemap.descendants().slice(0))
+    //   .join("tspan")
+    //     .attr("x", 3)
+    //     .attr("y", (d, i, D) => `${(i === D.length - 1) * 0.3 + 1.1 + i * 0.9}em`)
+    //     .attr("fill-opacity", (d, i, D) => i === D.length - 1 ? 0.7 : null)
+    //     .text(d => d.data.name);
+
+    });
+
+
+
+});
+// function to map pods to nodes
+async function mapPodsToNodes() {
+    const nodeData = await getNodes();
+    const podData = await getPods();
```

Comment:
If the requests are going to take time and are independent you can you the following to prevent them being waterfalled

`const [nodeData, podData] = await Promise.all([getNodes(), getPods()]);`